python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -182,7 +190,13 @@ def groupId(self):
     def commandName(self):
         return 'gdal_calc'
 
+    def processAlgorithm(self, parameters, context, feedback):
+        if GdalUtils.version() < 3030000 and self.EXTENT in parameters.keys():
+            raise QgsProcessingException(self.tr('A target extent parameter is present while running a gdal version below its minimum requirement (>= 3.3)'))
```

Comment:
How about just "The output extent option is only available on GDAL 3.3 or later"?